### PR TITLE
fix(notifications): Add Notifications to planner

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Add success, failure, and warning notifications to planner.
Notifications will be displayed for creation and modification of iterations and work items.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by pranavgore09** [comment]
This will be blocked by [core#PR](https://github.com/almighty/almighty-core/pull/1437)
I am trying to gather/analyze information to reach the solution.

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by hectorj2f** [comment]
[test]

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by hectorj2f** [comment]
[test]

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

File: src/app/components/iterations-modal/iterations-modal.component.ts [hunk 1]
```diff
@@ -6,13 +6,13 @@ import { Subscription } from 'rxjs/Subscription';
 import { cloneDeep } from 'lodash';
 import * as moment from 'moment';
 import { IMyOptions, IMyDateModel } from 'mydatepicker';
-import { Broadcaster } from 'ngx-base';
+import { Broadcaster, Notification, Notifications ,NotificationType } from 'ngx-base';
```
Comment (debloper) [inline src/app/components/iterations-modal/iterations-modal.component.ts]: Nitpick - spacing:
`Notifications ,NotificationType` -> `Notifications, NotificationType`

File: src/app/components/iterations-modal/iterations-modal.component.ts [hunk 2]
```diff
@@ -360,25 +376,48 @@ export class FabPlannerIterationModalComponent implements OnInit, OnDestroy, OnC
           this.iterationService.updateIteration(this.iteration)
             .subscribe((iteration) => {
               this.onSubmit.emit(iteration);
+              let toastIterationName = this.iteration.attributes.name;
+              if (toastIterationName.length > 15) {
+                toastIterationName = toastIterationName.slice(0, 15) + '...';
```
Comment (debloper) [inline src/app/components/iterations-modal/iterations-modal.component.ts]: This is a good workaround, but isn't it better to manage the overflow ellipsis in CSS?

Pseudo-code:
```html
<notification-container>
  <!-- overflow:ellipsis applied on this iteration-name element -->
  <iteration-name title/tooltip="iteration-name">iteration-name</iteration-name>
  has started...
</notification-container>
```

We're already crafting the DOM in next block, so why not just do it there, right?

File: src/app/components/iterations-modal/iterations-modal.component.ts [hunk 3]
```diff
@@ -360,25 +376,48 @@ export class FabPlannerIterationModalComponent implements OnInit, OnDestroy, OnC
...
+              if (this.modalType == 'start') {
+                let message = 'There is already a current iteration. You must close the current iteration before starting this one.';
+                try {
+                    this.notifications.message({
+                      message: message,
+                      type: NotificationType.WARNING
+
+                    } as Notification);
+                  } catch (e) {
+                    console.log('Error displaying notification. There is already a current iteration. You must close the current iteration before starting this one.')
```
Comment (debloper) [inline src/app/components/iterations-modal/iterations-modal.component.ts]: `console.log('Error displaying notification. ', message);`

File: src/app/components/iterations-modal/iterations-modal.component.ts [hunk 4]
```diff
@@ -360,25 +376,48 @@ export class FabPlannerIterationModalComponent implements OnInit, OnDestroy, OnC
...
+                  }
+              } else {
+                  this.eventService.showErrorModal.next({
+                    status: true,
+                    message: 'Error updating iteration.' + e
```
Comment (debloper) [inline src/app/components/iterations-modal/iterations-modal.component.ts]: Nitpick - spacing:
`message: 'Error updating iteration: ' + e` OR `message: 'Error updating iteration. ' + e`

File: src/app/components/overlay/overlay.component.scss [hunk 5]
```diff
@@ -0,0 +1,2 @@
+@import '../../../assets/stylesheets/base';
+.close {display: none;}
```
Comment (debloper) [inline src/app/components/overlay/overlay.component.scss]: Nitpick - sparse please:
`.close { display: none; }`

File: src/app/components/planner-list/planner-list.component.ts [hunk 6]
```diff
@@ -34,7 +34,7 @@ import {
 import { TreeNode } from 'angular2-tree-component';
 
 import { cloneDeep } from 'lodash';
-import { Broadcaster, Logger } from 'ngx-base';
+import { Broadcaster, Logger, Notification, Notifications ,NotificationType } from 'ngx-base';
```
Comment (debloper) [inline src/app/components/planner-list/planner-list.component.ts]: Same as before.

File: src/app/components/planner-list/planner-list.component.ts [hunk 7]
```diff
@@ -509,10 +515,31 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
...
           console.log('Added WI matches the applied filters');
+          let message = 'Work item ' + item.attributes['system.title'] + ' added.'
+          try {
+            this.notifications.message({
+              message: message,
+              type: NotificationType.SUCCESS
+
+            } as Notification);
+          } catch (e) {
+            console.log('Error displaying notification. Work item updated.')
```
Comment (debloper) [inline src/app/components/planner-list/planner-list.component.ts]: Added or updated? Better to append the `message` itself for consistency after the notification display error sentence, imo.

File: src/app/components/planner-list/planner-list.component.ts [hunk 8]
```diff
@@ -509,10 +515,31 @@ export class PlannerListComponent implements OnInit, AfterViewInit, DoCheck, OnD
...
+          //Notify user that the added WI will not be seen
+          let message = item.attributes['system.title'] + ' created. It does not match the filter criteria and is not displayed in this list. To view the work item, click on Backlog.'
+          try {
+            this.notifications.message({
+              message: message,
+              type: NotificationType.WARNING
+
+            } as Notification);
+          } catch (e) {
+            console.log('Error displaying notification. New iteration added.')
```
Comment (debloper) [inline src/app/components/planner-list/planner-list.component.ts]: Same thing here.

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 9]
```diff
@@ -591,11 +593,10 @@ export class WorkItemDetailComponent implements OnInit, OnDestroy {
...
         .catch((error: Error | any) => {
           this.saving = false;
-          this.savingError = true;
-          this.errorMessage = 'Something went wrong. Try again.'
-          if (error && error.status && error.statusText) {
-            this.errorMessage = error.status + ' : ' + error.statusText+ '. Try again.';
-          }
+          this.eventService.showErrorModal.next({
+            status: true,
+            message: 'Error creating work item.' + error
```
Comment (debloper) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: Same as before (spacing for readability).

File: src/app/components/work-item-quick-add/work-item-quick-add.component.ts [hunk 10]
```diff
@@ -227,8 +227,13 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, OnChanges,
         },
         error => {
           this.error = error;
-          this.qaSubmit.nativeElement.removeAttribute('disabled');
-          this.qaTitle.nativeElement.removeAttribute('disabled');
+          this.resetQuickAdd();
+          this.eventService.showErrorModal.next({
+              status: true,
+              message: 'Error creating new work item.' + error
```
Comment (debloper) [inline src/app/components/work-item-quick-add/work-item-quick-add.component.ts]: Same as before (spacing for readability).

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2017-11 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2017-12 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee fabric8/fabric8-ui:SNAPSHOT-PR-2017-13 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2017-fabric8-planner.badger.fabric8.io

**Pull request closed without merging**